pkg/api/validation/validation.go

```diff
@@ -2426,7 +2405,7 @@ func ValidatePreferredSchedulingTerms(terms []api.PreferredSchedulingTerm, fldPa
 // validatePodAffinityTerm tests that the specified podAffinityTerm fields have valid data
 func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopologyKey bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if !utilfeature.DefaultFeatureGate.Enabled(features.AffinityInAnnotations) && len(podAffinityTerm.TopologyKey) == 0 {
+	if len(podAffinityTerm.TopologyKey) == 0 {
```

Comment:
You wrote this check iirc. I tried removing this, but there were tests that depended on it.  
Refactoring should probably go into a separate PR.